src/lib/mailer/index.js

```diff
@@ -0,0 +1,198 @@
+      } = report;
+      const collaboratorEmailAddresses = collaborators.map((c) => c.email);
+      const reportPath = path.join(process.env.TTA_SMART_HUB_URI, 'activity-reports', String(id));
+      const email = new Email({
+        message: {
+          from: FROMEMAILADDRESS,
+        },
+        send,
+        transport,
+        htmlToText: {
+          wordwrap: 120,
+        },
+      });
+      return email.send({
+        template: path.resolve(emailTemplatePath, 'report_approved'),
+        message: {
+          to: [author.email, ...collaboratorEmailAddresses],
+        },
+        locals: {
+          manager: approvingManager.name,
+          reportPath,
+          displayId,
+        },
+      });
+    } catch (err) {
+      logger.error(err);
+    }
+  }
+  return Promise.resolve(null);
+};
+
+export const managerApprovalRequested = (report, transport = defaultTransport) => {
+// Set these inside the function to allow easier testing
+  const { FROMEMAILADDRESS, SENDNOTIFICATIONS } = process.env;
+  if (SENDNOTIFICATIONS === 'true') {
+    try {
+      const {
+        id,
+        author,
+        displayId,
+        approvingManager,
+      } = report;
+      const reportPath = path.join(process.env.TTA_SMART_HUB_URI, 'activity-reports', String(id));
+      const email = new Email({
+        message: {
+          from: FROMEMAILADDRESS,
+        },
+        send,
+        transport,
+        htmlToText: {
+          wordwrap: 120,
+        },
+      });
+      return email.send({
+        template: path.resolve(emailTemplatePath, 'manager_approval_requested'),
+        message: {
+          to: [approvingManager.email],
+        },
+        locals: {
+          author: author.name,
+          reportPath,
+          displayId,
+        },
+      });
+    } catch (err) {
+      logger.error(err);
+    }
+  }
+  return Promise.resolve(null);
+};
+
+export const notifyCollaborator = (report, newCollaborator, transport = defaultTransport) => {
+// Set these inside the function to allow easier testing
+  const { FROMEMAILADDRESS, SENDNOTIFICATIONS } = process.env;
+  if (SENDNOTIFICATIONS === 'true') {
+    try {
+      const {
+        id,
+        displayId,
+      } = report;
+      const reportPath = path.join(process.env.TTA_SMART_HUB_URI, 'activity-reports', String(id));
+      const email = new Email({
+        message: {
+          from: FROMEMAILADDRESS,
+        },
+        send,
+        transport,
+        htmlToText: {
+          wordwrap: 120,
+        },
+      });
+      return email.send({
+        template: path.resolve(emailTemplatePath, 'collaborator_added'),
+        message: {
+          to: [newCollaborator.email],
+        },
+        locals: {
+          reportPath,
+          displayId,
+        },
+      });
+    } catch (err) {
+      logger.error(err);
+    }
+  }
+  return Promise.resolve(null);
+};
+
+export const collaboratorAdded = (report, newCollaborators, transport = defaultTransport) => {
```

Comment:
Should we lump all these collaborators via `bcc` to just send one email? I remember at my old place we lowered our email quote by lumping them together (though, we're not sending as many emails as there)

Reply:
My preference would be to leave it as is for now for simplicity's sake. If we start running into any quota limits, we can keep this in our back pocket, but I am guessing that will be unlikely given the number of users this system will support. We will have a better idea about constraints when we actually find out about the smtp gateway we will be using.